src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageParser.cs

```diff
@@ -0,0 +1,193 @@
+
+                        if (_data.Count > 0)
+                        {
+                            //Find the final size of the payload
+                            var payloadSize = 0;
+                            foreach (var dataLine in _data)
+                            {
+                                payloadSize += dataLine.Length;
+                            }
+                            var payload = new byte[payloadSize];
+
+                            //Copy the contents of the data array to a single buffer
+                            var offset = 0;
+                            foreach (var dataLine in _data)
+                            {
+                                dataLine.CopyTo(payload, offset);
+                                offset += dataLine.Length;
+                            }
+
+                            message = new Message(payload, messageType);
+                        }
+                        else
+                        {
+                            //Empty message
+                            message = new Message(Array.Empty<byte>(), messageType);
+                        }
+
+                        consumed = buffer.End;
+                        return ParseResult.Completed;
+                }
+
+                //Peek into next byte. If it is a carriage return byte, then advance to the next state
+                if (reader.Peek() == ByteCR)
+                {
+                    _internalParserState = InternalParseState.ReadEndOfMessage;
+                }
+            }
+            return ParseResult.Incomplete;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private Span<byte> ConvertBufferToSpan(ReadableBuffer buffer)
+        {
+            if (buffer.IsSingleSpan)
+            {
+                return buffer.First.Span;
+            }
+            return buffer.ToArray();
+        }
+
+        public void Reset()
+        {
+            _internalParserState = InternalParseState.ReadMessageType;
+            _data.Clear();
+        }
+
+        private bool StartsWithDataPrefix(ReadOnlySpan<byte> line)
+        {
+            var dataPrefix = Encoding.UTF8.GetBytes("data: ");
+            var prefixSpan = new ReadOnlySpan<byte>(dataPrefix);
+            if (line.StartsWith(dataPrefix))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private MessageType GetMessageType(ReadOnlySpan<byte> line)
+        {
+            if (!StartsWithDataPrefix(line))
+            {
+                throw new FormatException("Expected the message prefix 'data: '");
+            }
+
+            if (line.Length != 7)
+            {
+                throw new FormatException("There was an error parsing the message type");
+            }
+
+            //Skip the "data: " part of the line
+            var type = (char)line[6];
+            switch (type)
+            {
+                case 'T':
```

Comment:
Did you intend to use TextTypeFlag etc. and therefore made them static readonly?